src/tls_stream.rs

```diff
@@ -501,6 +502,30 @@ impl<S> TlsStream<S>
         };
 
         unsafe {
+            // check if we trust the root-CA explicitly
+            let mut para_flags = winapi::CERT_CHAIN_POLICY_IGNORE_ALL_REV_UNKNOWN_FLAGS;
+            if let Some(ref mut store) = self.cert_store {
+                let cert_chain = *cert_chain.0;
+                if cert_chain.cChain > 0 {
+                    let first_rgp_chain = **cert_chain.rgpChain;
```

Comment:
seems to be exactly what OSCrypto and Microsoft is doing, doesn't it?
https://github.com/Microsoft/Windows-classic-samples/blob/master/Samples/Win7Samples/security/cryptoapi/VerifyNameTrust/VerifyNameTrust/VerifyNameTrust.cpp#L172